engine/src/main/java/org/terasology/world/block/family/DefaultBlockFamilyFactoryRegistry.java

```diff
@@ -16,27 +16,125 @@
+import org.terasology.world.block.BlockBuilderHelper;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
 
+import java.lang.reflect.InvocationTargetException;
 import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
 
-public class DefaultBlockFamilyFactoryRegistry implements BlockFamilyFactoryRegistry {
-    private Map<String, BlockFamilyFactory> registryMap = Maps.newHashMap();
-    private BlockFamilyFactory defaultBlockFamilyFactory = new SymmetricBlockFamilyFactory();
+public class DefaultBlockFamilyFactoryRegistry implements BlockFamilyRegistry {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultBlockFamilyFactoryRegistry.class);
+    private Context context;
 
-    public void setBlockFamilyFactory(String id, BlockFamilyFactory blockFamilyFactory) {
-        registryMap.put(id.toLowerCase(), blockFamilyFactory);
+
+    private Map<String, Class<? extends AbstractBlockFamily>> registryMap = Maps.newHashMap();
+
+    public DefaultBlockFamilyFactoryRegistry(Context context) {
+        this.context = context;
     }
 
-    @Override
-    public BlockFamilyFactory getBlockFamilyFactory(String blockFamilyFactoryId) {
-        if (blockFamilyFactoryId == null || blockFamilyFactoryId.isEmpty()) {
-            return defaultBlockFamilyFactory;
+    public static BlockFamily createFamily(Class<? extends AbstractBlockFamily> blockFamily, BlockFamilyDefinition blockFamilyDefinition, BlockBuilderHelper blockBuilderHelper) {
+        try {
+            SimpleClassFactory simpleClassFactory = new SimpleClassFactory(new ParameterProvider() {
+                @Override
+                public <T> Optional<T> get(Class<T> type) {
+                    if (type.isAssignableFrom(BlockBuilderHelper.class)) {
+                        return Optional.ofNullable((T) blockBuilderHelper);
+                    } else if (type.isAssignableFrom(BlockFamilyDefinition.class)) {
+                        return Optional.ofNullable((T) blockFamilyDefinition);
+                    }
+                    return Optional.empty();
+                }
+            });
+            BlockFamily result = simpleClassFactory.instantiateClass(blockFamily).get();
+            InjectionHelper.inject(result);
+            if(result.getURI() == null)
+                throw new Exception("Family Is missng a BlockUri");
+
+            return result;
+        } catch (Exception e) {
+            logger.error("Failed to load blockFamily {}", blockFamily, e);
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static BlockFamily createFamily(Class<? extends AbstractBlockFamily> blockFamily, BlockFamilyDefinition blockFamilyDefinition, BlockShape shape, BlockBuilderHelper blockBuilderHelper) {
+        try {
+            SimpleClassFactory simpleClassFactory = new SimpleClassFactory(new ParameterProvider() {
```

Comment:
This anonymous class smells funny. I think you could refactor this into a new class (with its own file) and take the parameters in this method (bound here by closure...) in a constructor and clean this up a bit.